pystac/extensions/file.py

```diff
@@ -99,6 +99,16 @@ class FileExtension(
        >>> file_ext = FileExtension.ext(asset)
     """
 
+    asset_href: str
+    """The ``href`` value of the :class:`~pystac.Asset` being extended."""
+
+    properties: Dict[str, Any]
+    """The :class:`~pystac.Asset` fields, including extension properties."""
+
+    additional_read_properties: Optional[Iterable[Dict[str, Any]]] = None
```

Comment:
Why is this Iterable?

Reply:
Good question, I'm actually not sure where that came from. I'll change that to `Optional[Dict[str, Any]]` for all of the extensions.

Reply:
Actually, it looks like this was introduced prior to this PR, probably as part of #309. @lossyrob Maybe you can clarify what the intent was here. The goal of `additional_read_properties` seems to be to fall back to reading a property from the owning Item in the case where an Asset does not have that property. However, it seems like this could be accomplished by typing this as `Optional[Dict[str, Any]]` and doing something like `self.additional_read_properties = item.properties` instead of `self.additional_read_properties = [item.properties]`. Are there any cases where we would want to pass in more than 1 dictionary for this?

Reply:
This was to account for the situation where you'd want to have multiple levels of property fallbacks - e.g. if there was something on the Asset, Item or Collection summary, or some set of defaults. I don't believe it covers a current need, but I wanted to future proof it a bit given that we've had this situation in the past (both Item properties and Collection properties (rip) were able to define values)